src/jdk.httpserver/share/classes/sun/net/httpserver/ServerImpl.java

```diff
@@ -733,7 +734,17 @@ public void run() {
                     connection.raw = rawin;
                     connection.rawout = rawout;
                 }
-                Request req = new Request(rawin, rawout);
+
+                Request req;
+                try {
+                    req = new Request(rawin, rawout, newconnection && !https);
```

Comment:
I'd skip the `!https` check here.

Reply:
We only need the check for clear connections. Why would you keep it with https?